apps/site/src/app/design-system/[brand]/[...component]/components/content-tabs/components/intro/components/table-of-contents/table-of-contents.component.tsx

```diff
@@ -36,18 +36,45 @@ const HEADER_HEIGHT = {
 
 const BREAKPOINT_MD = 768;
 
-function Link({ href, children }: { children?: React.ReactNode; href?: string }) {
+function Link({ href, children }: { children?: React.ReactNode; href: string }) {
+  const router = useRouter();
+  const pathname = usePathname();
+  const [scrollToHash, setScrollToHash] = useState<string | null>(null);
+  const [path, hash] = href.split('#');
+
+  const scrollToSection = useCallback(() => {
+    const viewport = window.innerWidth < BREAKPOINT_MD ? 'sm' : 'lg';
+    const bodyRect = document.body.getBoundingClientRect();
+    const elemRect = document?.querySelector(`#${hash}`)?.getBoundingClientRect();
+    const offset = (elemRect?.top || 0) - bodyRect.top - HEADER_HEIGHT[viewport];
+
+    window?.scrollTo({ top: offset, behavior: 'smooth' });
+    window.history.pushState(null, '', `#${hash}`);
+  }, [hash]);
+
+  useEffect(() => {
+    if (scrollToHash) {
+      setTimeout(() => {
+        scrollToSection();
+      }, 500);
```

Comment:
minor: I would setup like `const DELAY_TIME_TO_SCROLL = 500;` outside not leave the magic numbers (for reference: https://eslint.org/docs/latest/rules/no-magic-numbers) .

Reply:
added this to the PR